programs/Makefile

```diff
@@ -52,6 +52,10 @@ CPPFLAGS  += -I$(ZSTDDIR)/legacy
 ZSTDLEGACY_FILES:= $(ZSTDDIR)/legacy/*.c
 endif
 
+ifeq ($(shell uname), FreeBSD)
+CPPFLAGS  += -D_POSIX_C_SOURCE=200112L
```

Comment:
I tend to prefer such detection macro to be set from source code itself. My understanding is that this macro should be enabled from `utils.h`. Is there a safe way to do so ?
cc @inikep , he has some experience on this topic.